nltk/translate/bleu_score.py

```diff
@@ -166,13 +160,12 @@ def corpus_bleu(list_of_references, hypotheses, weights=[0.25, 0.25, 0.25, 0.25]
     # Calculate corpus-level brevity penalty.
     bp = _brevity_penalty(ref_lengths, hyp_lengths)
     
-    # Calculate corpus-level modified precision.
-    p_n = []
-    for i, w in enumerate(weights, start=1):
-        pn = p_numerators[i] / p_denominators[i]
-        p_n.append(w* math.log(pn))
+    # Calculate sum of corpus-level modified precisions.
+    s = (0 if (p_numerators[i] | p_denominators[i]) == 0 else
```

Comment:
what does `(p_numerators[i] | p_denominators[i]) == 0` mean? does it check whether `p_numerators[i]` or `p_denominators[i]` is 0?

if yes, i would rewrite to something like `0 if not(p_numerators[i] and p_denominators[i]) else ...`